core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -848,15 +855,81 @@ public Graph build() {
+                LOGGER.warn("No FunctionAuthoriser hook was supplied, adding default hook.");
+                hooks.add(new FunctionAuthoriser(FunctionAuthoriserUtil.DEFAULT_UNAUTHORISED_FUNCTIONS));
+            }
+        }
+
+        private void updateNamedViewResolverHook(final GraphConfig config) {
+            updateGetFromCacheHookWhenMissingOrHasWrongSuffix(
+                    config,
+                    AddNamedView.class,
+                    NamedViewResolver.class,
+                    null != properties ? properties.getCacheServiceNamedViewSuffix(config.getGraphId()) : null);
+        }
+
+        private void updateNamedOperationResolverHook(final GraphConfig config) {
+            updateGetFromCacheHookWhenMissingOrHasWrongSuffix(
+                    config,
+                    AddNamedOperation.class,
+                    NamedOperationResolver.class,
+                    null != properties ? properties.getCacheServiceNamedOperationSuffix(config.getGraphId()) : null);
+        }
+
+        /**
+         * This method is designed to keep suffix used for cache names aligned between
+         * a Handler which Adds to the cache and ResolverHook which Gets from the cache.
+         * If the suffix is mismatched then writing and reading to cache will not behave correctly.
+         *
+         * @param config           the graphConfig containing the Hooks
+         * @param operationClass   the Operation requiring cache write
+         * @param hookClass        the Hook requiring cache reading
+         * @param suffixFromProperties the suffix from property
+         */
+        private void updateGetFromCacheHookWhenMissingOrHasWrongSuffix(final GraphConfig config, final Class<? extends Operation> operationClass, final Class<? extends GetFromCacheHook> hookClass, final String suffixFromProperties) {
+            if (store.isSupported(operationClass)) {
+                //Get Handler
+                final OperationHandler<Operation> addToCacheHandler = store.getOperationHandler(operationClass);
+                //If Handler is a AddToCacheHandler
+                final String suffix;
+                if (AddToCacheHandler.class.isAssignableFrom(addToCacheHandler.getClass())) {
+                    // get Suffix
+                    suffix = ((AddToCacheHandler<?>) addToCacheHandler).getSuffixCacheName();
+                } else {
+                    // otherwise get from properties
+                    LOGGER.warn(String.format(HANDLER_WAS_NOT_EXPECTED_TYPE_ADD_TO_CACHE_HANDLER, operationClass, AddToCacheHandler.class.getSimpleName(), suffixFromProperties));
+                    suffix = suffixFromProperties;
+                }
+                final List<GraphHook> hooks = config.getHooks();
+                //Is GetFromCacheHook missing
+                if (!hasHook(hooks, hookClass)) {
+                    //Warn about the mistake and add a resolver
```

Comment:
A mistake has happend.
The graph is being built with incomplete configuration. 
That's because all those test a not configured correctly. They mostly don't care, because nearly all those graphs aren't using NamedOp's but the AddNamedOp is still configured.

The correct test (in a perfect world) would use a Store which only contains the handlers required for the tests.

Reply:
because of the importance of cache. Before hand there was only ever 1 cache. Now there is option to have different caches or shared caches. So now I think a warning has to be given that a cache is being made with assumption of the correct name to be used.

In the example recent of migration to v2 we had issues with this stuff. I can switch it to info, but no one read info :D

Reply:
If the documentation and getting started guides are first updated to make it clear that users are expected to supply boilerplate hooks with cache suffixes, then this could be a warning. As it is now, users are not expected to supply this and will be confused by this warning. It should be info level.